lib/src/participant/local.dart

```diff
@@ -1146,3 +1151,183 @@ extension RPCMethods on LocalParticipant {
+          ?.call(totalProgress.toDouble() / (fileIds?.length ?? 1));
+    }
+
+    final writer = await streamText(StreamTextOptions(
+      streamId: streamId,
+      totalSize: totalTextLength,
+      destinationIdentities: options?.destinationIdentities ?? [],
+      topic: options?.topic,
+      attachedStreamIds: fileIds ?? [],
+    ));
+
+    await writer.write(text);
+    // set text part of progress to 1
+    handleProgress(1, 0);
+
+    await writer.close();
+
+    if (options?.attachments != null) {
+      var idx = 0;
+      await Future.wait<void>(
+        options?.attachments.map(
+              (file) {
+                var curIdx = idx++;
+                return _sendFile(
+                  fileIds![curIdx],
+                  file,
+                  SendFileOptions(
+                      topic: options.topic,
+                      mimeType: mime(basename(file.path)),
+                      onProgress: (progress) {
+                        handleProgress(progress, curIdx + 1);
+                      }),
+                );
+              },
+            ).toList() ??
+            [],
+      );
+    }
+    return writer.info;
+  }
+
+  Future<TextStreamWriter> streamText(StreamTextOptions? options) async {
+    final streamId = options?.streamId ?? Uuid().v4();
+
+    final info = TextStreamInfo(
+      id: streamId,
+      mimeType: 'text/plain',
+      timestamp: DateTime.now().millisecondsSinceEpoch,
+      topic: options?.topic ?? '',
+      size: options?.totalSize ?? 0,
+    );
+
+    final header = lk_models.DataStream_Header(
+      streamId: streamId,
+      mimeType: info.mimeType,
+      topic: info.topic,
+      timestamp: Int64(info.timestamp),
+      totalLength: Int64(options?.totalSize ?? 0),
+      textHeader: lk_models.DataStream_TextHeader(
+        version: options?.version,
+        attachedStreamIds: options?.attachedStreamIds,
+        replyToStreamId: options?.replyToStreamId,
+        operationType: options?.type == 'update'
+            ? lk_models.DataStream_OperationType.UPDATE
+            : lk_models.DataStream_OperationType.CREATE,
+      ),
+    );
+    final destinationIdentities = options?.destinationIdentities;
+    final packet = lk_models.DataPacket(
+      destinationIdentities: destinationIdentities,
+      streamHeader: header,
+    );
+    await room.engine.sendDataPacket(packet, reliability: true);
+
+    final writableStream = StreamWriterImpl(
+        destinationIdentities: destinationIdentities!,
+        engine: room.engine,
+        streamId: streamId);
+
+    onEngineClose() async {
+      await writableStream.close();
+    }
+
+    var cancelFun =
+        room.engine.events.once<EngineClosingEvent>((_) => onEngineClose);
+
+    final writer = TextStreamWriter(
+      writableStream: writableStream,
+      info: info,
+      onClose: () {
+        cancelFun?.call();
+      },
+    );
+
+    return writer;
+  }
+
+  Future<Map<String, String>> sendFile(
```

Comment:
I recently added also the `streamBytes` API to JS. 
This adds consistency across other SDKs and allows for `sendFile` to be a light wrapper around `streamBytes`. see https://github.com/livekit/client-sdk-js/commit/c0c5642ede2d7e202e9d03916d74fb6980d4d48f